kof-operator/internal/controller/clusterdeployment_kof_cluster_child_role.go

```diff
@@ -129,11 +116,75 @@ func (c *ChildClusterRole) GetRegionalConfigMap() (*RegionalClusterConfigMap, er
 	return NewRegionalClusterConfigMap(c.ctx, regionalClusterConfigMap, c.client)
 }
 
-func (c *ChildClusterRole) DiscoverRegionalClusterConfigMapByLocation(
-	crossNamespace bool,
-) (*corev1.ConfigMap, error) {
+func (c *ChildClusterRole) DiscoverRegionalClusterCmByLabel(regionalClusterName string) (*corev1.ConfigMap, error) {
+	ok := false
+	log := log.FromContext(c.ctx)
+	crossNamespace := os.Getenv("CROSS_NAMESPACE") == "true"
+	regionalClusterNamespace := c.clusterNamespace
+	regionalClusterConfigMap := new(corev1.ConfigMap)
+
+	if crossNamespace {
+		regionalClusterNamespace, ok = c.clusterDeployment.Labels[KofRegionalClusterNamespaceLabel]
+		if !ok {
+			err := fmt.Errorf(`"%s" label is required`, KofRegionalClusterNamespaceLabel)
+			log.Error(
+				err, fmt.Sprintf(`when crossNamespace is true and "%s" label is set`, KofRegionalClusterNameLabel),
+				"crossNamespace", crossNamespace,
+				KofRegionalClusterNameLabel, regionalClusterName,
+				"childClusterDeploymentName", c.clusterName,
+			)
+			return nil, err
+		}
+	}
+
+	if c.isClusterInRegion {
+		isSameRegion, err := k8s.IsClusterInSameKcmRegion(c.ctx, c.client,
+			c.clusterName, c.clusterNamespace,
+			regionalClusterName, regionalClusterNamespace,
+		)
+
+		if err != nil {
+			err := fmt.Errorf("failed to determine if clusters are in the same KCM region: %v", err)
+			log.Error(
+				err, "when crossNamespace is true and regional cluster label is set",
```

Comment:
crossNamespace is not always true here:

```suggestion
				err, "when regional cluster label is set",
```